backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/hooks/StatusBranchHookExecutor.java

```diff
@@ -108,9 +115,30 @@ private Option<String> executeHookFor(String branchName) throws GitMacheteExcept
     }
   }
 
+  public static String hashFile(String algorithm, File f) throws IOException, NoSuchAlgorithmException {
+    MessageDigest md = MessageDigest.getInstance(algorithm);
+
+    try (BufferedInputStream in = new BufferedInputStream(new FileInputStream(f));
+        DigestOutputStream out = new DigestOutputStream(OutputStream.nullOutputStream(), md)) {
+      in.transferTo(out);
+    }
+
+    return new BigInteger(1, md.digest()).toString(16);
```

Comment:
As the meaning of 1 is sheerly unclear
```suggestion
    return new BigInteger(/* signum */ 1, md.digest()).toString(16);
```